overlord/snapstate/snapstate_update_test.go

```diff
@@ -10287,6 +10236,334 @@ func (s *snapmgrTestSuite) TestAutoRefreshCreatePreDownload(c *C) {
+		refreshAppsCheckCalled++
+		return snapstate.NewBusySnapError(si, []int{123}, nil, nil)
+	})
+	defer restore()
+
+	// Never close monitoring channel to avoid triggering a continued auto-refresh
+	var monitored int
+	restore = snapstate.MockCgroupMonitorSnapEnded(func(string, chan<- string) error {
+		monitored++
+		return nil
+	})
+	defer restore()
+
+	// Setup test snaps
+	s.state.Lock()
+	snapstate.Set(s.state, "some-snap", &snapstate.SnapState{
+		Active: true,
+		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{
+			{RealName: "some-snap", SnapID: "some-snap-id", Revision: snap.R(1)},
+		}),
+		Current:  snap.R(1),
+		SnapType: string(snap.TypeApp),
+	})
+	// Verify list is empty
+	checkLastRecordedInhibitedSnaps(c, s.state, nil)
+	s.state.Unlock()
+
+	snapstate.CanAutoRefresh = func(*state.State) (bool, error) { return true, nil }
+	// Trigger autorefresh.Ensure().
+	c.Assert(s.snapmgr.Ensure(), IsNil)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// First auto-refresh attempt
+	s.settle(c)
+	chgs := s.state.Changes()
+	c.Assert(chgs, HasLen, 1)
+	c.Check(chgs[0].Kind(), Equals, "pre-download")
+	c.Check(chgs[0].Status(), Equals, state.DoneStatus)
+	// If all refresh candidates are blocked from auto-refresh and only a
+	// pre-download change is created for those snaps we should still record
+	// a notice.
+	checkRefreshInhibitNotice(c, s.state, 1)
+	checkLastRecordedInhibitedSnaps(c, s.state, []string{"some-snap"})
+
+	forceAutoRefresh := func() {
+		s.state.Unlock()
+		// Fake nextRefresh to now.
+		s.snapmgr.MockNextRefresh(time.Now())
+		// Fake that the retryRefreshDelay is over
+		restore = snapstate.MockRefreshRetryDelay(1 * time.Millisecond)
+		defer restore()
+		time.Sleep(10 * time.Millisecond)
+		// Trigger autorefresh.Ensure().
+		c.Assert(s.snapmgr.Ensure(), IsNil)
+		s.state.Lock()
+	}
+
+	// Force another auto-refresh to check that notice is not repeated
+	// for the same set of inhibited snaps.
+	forceAutoRefresh()
+	s.settle(c)
+
+	chgs = s.state.Changes()
+	c.Assert(chgs, HasLen, 2)
+	// Old pre-download
+	c.Check(chgs[0].Kind(), Equals, "pre-download")
+	c.Check(chgs[0].Status(), Equals, state.DoneStatus)
+	// New pre-download
+	c.Check(chgs[1].Kind(), Equals, "pre-download")
+	c.Check(chgs[1].Status(), Equals, state.DoneStatus)
+	// No new notices recorded because the set of inhibited snaps was not changed.
+	checkRefreshInhibitNotice(c, s.state, 1)
+	checkLastRecordedInhibitedSnaps(c, s.state, []string{"some-snap"})
+
+	// Add another inhibited snap.
+	snapstate.Set(s.state, "some-other-snap", &snapstate.SnapState{
+		Active: true,
+		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{
+			{RealName: "some-other-snap", SnapID: "some-other-snap-id", Revision: snap.R(1)},
+		}),
+		Current:  snap.R(1),
+		SnapType: string(snap.TypeApp),
+	})
+
+	// Force another auto-refresh to check that notice is recorded becuase
```

Comment:
As the misspell checker mentions, `s/becuase/because/` :)

Reply:
Thank you! updated